lib/ohai/plugins/linux/filesystem2.rb

```diff
@@ -0,0 +1,218 @@
+#
+# Author:: Phil Dibowitz <phil@ipom.com>
+# Author:: Adam Jacob <adam@chef.io>
+# Copyright:: Copyright (c) 2008 Opscode, Inc.
+# Copyright:: Copyright (c) 2015 Facebook, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+Ohai.plugin(:Filesystem2) do
+  provides "filesystem2"
+
+  def find_device(name)
+    %w{/dev /dev/mapper}.each do |dir|
+      path = File.join(dir, name)
+      return path if File.exist?(path)
+    end
+    name
+  end
+
+  def parse_line(line, have_lsblk)
+    if have_lsblk
+      regex = /NAME="(\S+)" UUID="(\S*)" LABEL="(\S*)" FSTYPE="(\S*)"/
+      if line =~ regex
+        dev = $1
+        dev = find_device(dev) unless dev.start_with?('/')
+        uuid = $2
+        label = $3
+        fs_type = $4
+        return {:dev => dev, :uuid => uuid, :label => label, :fs_type => fs_type}
+      end
+    else
+      bits = line.split
+      dev = bits.shift.split(':')[0]
+      f = {:dev => dev}
+      bits.each do |keyval|
+        if keyval =~ /(\S+)="(\S+)"/
+          key = $1.downcase.to_sym
+          key = :fs_type if key == :type
+          f[key] = $2
+        end
+      end
+      return f
+    end
+    return nil
+  end
+
+  def generate_device_view(fs)
+    view = {}
+    fs.each_value do |entry|
+      view[entry[:device]] = Mash.new unless view[entry[:device]]
+      entry.each do |key, val|
+        next if ['device', 'mount'].include?(key)
+        view[entry[:device]][key] = val
+      end
+      if entry[:mount]
+        view[entry[:device]][:mounts] = [] unless view[entry[:device]][:mounts]
+        view[entry[:device]][:mounts] << entry[:mount]
+      end
+    end
+    view
+  end
+
+  def generate_mountpoint_view(fs)
+    view = {}
+    fs.each_value do |entry|
+      next unless entry[:mount]
+      view[entry[:mount]] = Mash.new unless view[entry[:mount]]
+      entry.each do |key, val|
+        next if key == 'mount'
+        view[entry[:mount]][key] = val
+      end
+    end
+    view
+  end
+
+  collect_data(:linux) do
+    fs = Mash.new
+
+    # Grab filesystem data from df
+    so = shell_out("df -P")
+    so.stdout.lines do |line|
```

Comment:
I've been looking at enabling warnings for our tests and this usage emits a warning for being deprecated; it's suggested to use `each_line` instead. That said, there's a fair bit of cleanup to do to make the code warning free (including fixing the same usage elsewhere) so it's no big deal to leave it in.